source/AI.cpp

```diff
@@ -811,7 +815,8 @@ void AI::AskForHelp(Ship &ship, bool &isStranded, const Ship *flagship)
 bool AI::CanHelp(const Ship &ship, const Ship &helper, const bool needsFuel)
 {
 	// Fighters, drones, and disabled / absent ships can't offer assistance.
-	if(helper.CanBeCarried() || helper.GetSystem() != ship.GetSystem() || helper.Cloaking() == 1.
+	if(helper.CanBeCarried() || helper.GetSystem() != ship.GetSystem()
+			|| (helper.Cloaking() == 1. && helper.GetGovernment() != ship.GetGovernment())
```

Comment:
Why this change? It seems reasonable to let, for example, a Merchant ask for fuel from the Navy.

Reply:
That should still be possible - rather than be unable to ask any cloaked ship for help (current behavior), I make it so that a cloaked ship can be asked for help if that cloaked ship is of the same government